### PR TITLE
[v3.7] community/php7: security upgrade to 7.1.30

### PR DESCRIPTION
**Pull request by andypost**
https://www.php.net/ChangeLog-7.php#7.1.29
http://php.net/archive/2019.php#id2019-03-07-3
http://php.net/archive/2018.php#id2018-12-06-4
http://php.net/archive/2018.php#id2018-09-13-4

### CODE REVIEW

**Comment by andypost** [comment]
x86_64
```
=====================================================================
FAILED TEST SUMMARY
---------------------------------------------------------------------
Bug #76249 (stream filter convert.iconv leads to infinite loop on invalid sequence) [ext/iconv/tests/bug76249.phpt]
Test posix_getpwuid() function : basic functionality [ext/posix/tests/posix_getpwuid_basic.phpt]
=====================================================================
```

**Comment by andypost** [comment]
arm8
```
=====================================================================
FAILED TEST SUMMARY
---------------------------------------------------------------------
Bug #76249 (stream filter convert.iconv leads to infinite loop on invalid sequence) [ext/iconv/tests/bug76249.phpt]
IntlTimeZone::getErrorCode/Message(): basic test [ext/intl/tests/timezone_getErrorCodeMessage_basic.phpt]
IntlTimeZone::getOffset(): errors [ext/intl/tests/timezone_getOffset_error.phpt]
OO API [ext/tidy/tests/020.phpt]
=====================================================================
```

**Comment by ncopa** [comment]
what is the status here? can we merge this?

**Comment by andypost** [comment]
not yet, I need to check why tests fail

**Comment by andypost** [comment]
new version fails on the same tests
```
FAILED TEST SUMMARY
---------------------------------------------------------------------
Bug #76249 (stream filter convert.iconv leads to infinite loop on invalid sequence) [ext/iconv/tests/bug76249.phpt]
Test posix_getpwuid() function : basic functionality [ext/posix/tests/posix_getpwuid_basic.phpt]
```

**Comment by andypost** [comment]
on arm8
```
=====================================================================
FAILED TEST SUMMARY
---------------------------------------------------------------------
Bug #76249 (stream filter convert.iconv leads to infinite loop on invalid sequence) [ext/iconv/tests/bug76249.phpt]
IntlTimeZone::getErrorCode/Message(): basic test [ext/intl/tests/timezone_getErrorCodeMessage_basic.phpt]
IntlTimeZone::getOffset(): errors [ext/intl/tests/timezone_getOffset_error.phpt]
Test posix_getpwuid() function : basic functionality [ext/posix/tests/posix_getpwuid_basic.phpt]
=====================================================================
```

**Comment by andypost** [comment]
@ncopa The iconv tests fail because of musl implementation https://github.com/gliderlabs/docker-alpine/issues/216

File: community/php7/APKBUILD [hunk 1]
```diff
@@ -181,6 +181,14 @@ ppc64le) options="$options !check";;
 esac
 
 # secfixes:
+#   7.1.26-r0:
+#     - CVE-2019-6977
+#     - CVE-2018-19935
+#     - CVE-2018-19518
+#     - CVE-2018-17082
+#     - CVE-2018-14883
+#     - CVE-2018-14851
```
Comment (J0WI) [inline community/php7/APKBUILD]:
```suggestion
#     - CVE-2018-14851
#     - CVE-2018-7584
#     - CVE-2018-5712
```

File: community/php7/APKBUILD [hunk 2]
```diff
@@ -181,6 +181,14 @@ ppc64le) options="$options !check";;
 esac
 
 # secfixes:
+#   7.1.26-r0:
+#     - CVE-2019-6977
```
Comment (J0WI) [inline community/php7/APKBUILD]:
```suggestion
#     - CVE-2019-6977
#     - CVE-2018-20783
```

File: community/php7/APKBUILD [hunk 3]
```diff
@@ -181,6 +181,14 @@ ppc64le) options="$options !check";;
...
+#     - CVE-2018-19935
+#     - CVE-2018-19518
```
Comment (J0WI) [inline community/php7/APKBUILD]:
```suggestion
#     - CVE-2018-19518
#     - CVE-2018-19396
#     - CVE-2018-19395
```

File: community/php7/APKBUILD [hunk 4]
```diff
@@ -181,6 +181,14 @@ ppc64le) options="$options !check";;
...
+#     - CVE-2018-17082
```
Comment (J0WI) [inline community/php7/APKBUILD]:
```suggestion
#     - CVE-2018-17082
#     - CVE-2018-15132
#     - CVE-2018-14884
```

File: community/php7/APKBUILD [hunk 5]
```diff
@@ -181,6 +181,14 @@ ppc64le) options="$options !check";;
...
+#     - CVE-2018-14883
+#     - CVE-2018-14851
+#     - CVE-2016-10166
 #   7.1.17-r0:
```
Comment (J0WI) [inline community/php7/APKBUILD]:
```suggestion
#   7.1.17-r0:
#     - CVE-2018-10549
#     - CVE-2018-10548
#     - CVE-2018-10547
#     - CVE-2018-10546
#     - CVE-2018-10545
```

**Comment by J0WI** [comment]
Meanwhile 7.1.27 is out fixing
- CVE-2019-9637
- CVE-2019-9638
- CVE-2019-9639
- CVE-2019-9640
- CVE-2019-9641

**Comment by andypost** [comment]
@J0WI shere sre from this cves? I see no any in http://php.net/ChangeLog-7.php#7.1.27 and the same list of pending in https://github.com/alpinelinux/aports/pull/6574

**Comment by J0WI** [comment]
I got them directly from [MITRE](https://cve.mitre.org/), I think they have been requested by [Debian](https://www.debian.org/security/2019/dsa-4403). Same on #6574.

**Comment by andypost** [comment]
@J0WI thank a lot of secfixes, updated (guess does not skip something) also bumped to 7.1.27

**Comment by J0WI** [comment]
7.1.28 is out, pending CVE requests for https://bugs.php.net/bug.php?id=77753 and https://bugs.php.net/bug.php?id=77831.

**Comment by J0WI** [comment]
`CVE-2019-11034` and `CVE-2019-11035` have been assigned. I suggest to just disable the failing test on `arm8` in favor of the security fixes.

**Comment by andypost** [comment]
@ncopa does it makes sense to disable tests per arch? the version in 3.7 release fails some tests but builds and works fine

**Comment by andypost** [comment]
Also added new secfix https://bugs.php.net/bug.php?id=77950 2019-11036

**Pull request closed without merging**